GameConsole/screens/settingsMenu.h

```diff
@@ -0,0 +1,146 @@
+﻿#pragma once
+#include <sstream>
+
+#include "buttons.h"
+#include "IScreen.h"
+#include "transitionData.h"
+
+namespace screens
+{
+    class settingsMenu : public IScreen
+    {
+    private:
+        button buttons[4];
+        std::vector<button> playersButtons;
+        int currentButton = 0;
+        int currentPlayerButton = 0;
+
+        std::vector<std::string> players;
+        std::string configFilePath = "Data\\deck_setup.json";
+        int handCount = 7;
+        size_t seed = 12341234;
+        bool editingPlayers;
+
+        std::map<int, eventBus::delegate<transitionData>> transitionsMap = {
+            {
+                NAVIGATION_MAIN_MENU,
+                eventBus::delegate<transitionData>{std::bind(&settingsMenu::onHide, this, std::placeholders::_1)}
+            },
+            {
+                NAVIGATION_SETTINGS,
+                eventBus::delegate<transitionData>{std::bind(&settingsMenu::onShow, this, std::placeholders::_1)}
+            },
+            {
+                NAVIGATION_GAME,
+                eventBus::delegate<transitionData>{std::bind(&settingsMenu::onHide, this, std::placeholders::_1)}
+            },
+            {
+                NAVIGATION_GAME_OVER,
+                eventBus::delegate<transitionData>{std::bind(&settingsMenu::onHide, this, std::placeholders::_1)}
+            },
+        };
+
+        void setupPlayerButton();
+
+    public:
+        settingsMenu(std::shared_ptr<renderer::renderer> rdr, std::shared_ptr<eventBus::eventBus> events)
+            : IScreen(rdr, events)
+        {
+            for (std::pair<const int, eventBus::delegate<transitionData>> transitionMap : transitionsMap)
+            {
+                size_t id = events->subscribe<transitionData>(transitionMap.first, transitionMap.second.action);
+                transitionsMap[transitionMap.first].uid = id;
+            }
+
+            players.resize(2);
```

Comment:
Why 2?

Reply:
magic numbers gotta love them :), I'll create a const for it